app/models/api_user.rb

```diff
@@ -8,6 +8,10 @@ class ApiUser < ApplicationRecord
 
   has_secure_token :token, length: 36
 
+  default_scope { where(revoked_at: nil) }
```

Comment:
Shall we avoid using default scope given the documents?

Reply:
For future reference: adding the default scope is the pattern followed by gems like [`paranoia`](https://github.com/rubysherpas/paranoia) and [`soft_deletion`](https://github.com/grosser/soft_deletion) which was what I followed here. (Actually via [this medium article](https://medium.com/@cathmgarcia/soft-deletion-in-ruby-on-rails-a1d65d0172ab).)

I think it's probably safer when applied to a deleted/revoked state than for a fuzzier concept of what records to access, as we have in `document` — I think for documents it really is just that some views want to exclude some types and others don't, so default_scope is too blunt an instrument for it. We really would _almost never_ want to access a revoked key.

However, again, this is in practice probably one single method call in the authentication handler, so we're not gaining anything from the scopes anyway.